app/controllers/gems_adoptions_controller.rb

```diff
@@ -0,0 +1,30 @@
+class GemsAdoptionsController < ApplicationController
+  before_action :require_login
+
+  def new
+    @gem_adoption = params[:id]
+    @gem = GemsAdoption.new
+  end
+
+  def create
+    @gems_adoption = current_user.gems_adoptions.build(gems_adoption_params)
+    if @gems_adoption.save
+      redirect_to gem_path(id: @gems_adoption.gem)
+    else
+      flash[:error] = 'Error! Dont saved'
+      redirect_to :gems
```

Comment:
In case there is an error putting a gem up to adoption, we should `render :new` instead of redirecting the user to another view, in order to display the errors messages and give him the chance to fix the errors.